active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+
+The command can shared between different IInputValidationControls, as would most likely be the case in the scenario of a forms control.  A custom command can be made that listens for changes to all errors and knows when a form has been completed and ready for submition. The control will query the command as to whether they should perform validation.
+
+Developers can override any of this behavior by creating their own InputValidationCommand.
+ 
+ 
+    /// <summary>
+    /// Command that controls how an IInputValidationControl does validation.
+    /// </summary>
+    [webhosthidden]
+    unsealed runtimeclass InputValidationCommand
+    {
+        [method_name("CreateInstance")] InputValidationCommand();
+        Windows.UI.Xaml.Controls.InputValidationKind InputValidationKind;
+        /// <summary>
+        /// Called by a control to determine if it should call the Validate method.
+        /// </summary>
+        Boolean CanValidate(Windows.UI.Xaml.Controls.IInputValidationControl validationControl);
+        /// <summary>
+        /// If CanValidate returns true, Validate is invoked to perform validation on the
+        /// validationControl provided.
+        /// </summary>
+        void Validate(Windows.UI.Xaml.Controls.IInputValidationControl validationControl);
+        /// <summary>
+        /// Method that derived classes implement to provide their own implementation of the CanValidate method.
+        /// </summary>
+        overridable Boolean CanValidateCore(Windows.UI.Xaml.Controls.IInputValidationControl validationControl);
+        /// <summary>
+        /// Method that derived classes implement to provide their own implementation of the Validate method.
+        /// </summary>
+        overridable void ValidateCore(Windows.UI.Xaml.Controls.IInputValidationControl validationControl);
+    };
+
+## ValidationContext
+The `ValidationContext` class provides context as to what is being required by validation. This class is auto-generated by the XAML markup compiler in certain scenarios (see [IInputValidationControl]()). Controls can use the provided properties on the `ValidationContext` to customize their appearance.  For example, the IsInputRequired field is can allow controls to put the little * next to a header. 
+
+Note: Intended to be extended for things like IsPhoneNumber or IsEmail that will allow controls to have other behavior (like mask text box).
+
+    
+    /// <summary>
+    /// Provides more contextual information about how to validate, display, and/or format input.
+    /// </summary>
+    [webhosthidden]
+    unsealed runtimeclass InputValidationContext
+    {
+        [method_name("CreateInstance")] InputValidationContext(String memberName, Boolean isRequired);
+        Boolean IsInputRequired{ get; };
+        String MemberName{ get; };
+    };
+    
+## InputValidationError
```

Comment:
Can you explain why this is needed?

Why not just have `IInputValidationControl .ValidationErrors` return an `Windows.Foundation.Collections.IObservableVector<string>`?

Reply:
That's a great question, and i was wondering that myself a lot. A few reasons i kept it around:
1. Closer to what WPF has (maybe not a great reason?)
2. We might want to change the concept of an "error" to a "result" that can be either positive or negative, allowing controls to give visual feedback when a field is correctly filled out. This becomes a bit more complicated if we are relying on `INotifyDataErrorInfo` since it well, only notifies about errors :)
3. Easier to extend for the future if it turns out we want to add more functionality. This was a bigger deal when we were designing this feature as part of Windows, since we can't really change APIs once they ship. We probably can have more flexibility here and could just make this a string.

Reply:
There should probably be system brush resources added to the ThemeResource dictionaries for positive validation, as well as the negative error red and yellow, currently afforded